multimodal/health_multimodal/image/model/model.py

```diff
@@ -0,0 +1,167 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+
+from dataclasses import dataclass
+from typing import Any, List, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from pl_bolts.models.self_supervised.resnets import resnet18, resnet50
+from torch import Tensor as T
+
+from .modules import ClassifierHeadType, MLP, MTModel
+
+
+@dataclass
+class ImageModelOutput():
+    img_embedding: torch.Tensor
+    patch_embedding: torch.Tensor
+    projected_global_embedding: torch.Tensor
+    class_logits: torch.Tensor
+    projected_patch_embeddings: torch.Tensor
+
+
+class ImageModel(nn.Module):
+    """
+    Image encoder module
+    """
+
+    def __init__(self,
+                 img_model_type: str,
+                 joint_feature_size: int,
+                 freeze_encoder: bool = False,
+                 **downstream_classifier_kwargs: Any):
+        super().__init__()
+
+        # Initiate encoder, projector, and classifier
+        self.encoder = ImageEncoder(img_model_type)
+        self.feature_size = get_encoder_output_dim(self.encoder)
+        self.projector = MLP(self.feature_size, output_dim=joint_feature_size,
+                             hidden_dim=joint_feature_size, use_1x1_convs=True)
+        self.downstream_classifier_kwargs = downstream_classifier_kwargs
+        self.classifier = self.create_downstream_classifier() if downstream_classifier_kwargs else None
+
+        # Initialise the mode of modules
+        self.freeze_encoder = freeze_encoder
+        self.train()
+
+    def train(self, mode: bool = True) -> Any:
+        super().train(mode=mode)
+        '''
+        Called to switch the model between training and evaluation modes.
+        '''
```

Comment:
Should be

1. Enclosed in double quotes for consistency
1. In a single line
1. Before the call to `super`

Reply:
Fixed in 7665309.